intTests/test_mir_unsafe_assume_spec_statics/test.rs

```diff
@@ -0,0 +1,14 @@
+static mut A: u32 = 42;
+
+pub fn side_effect() -> u32 {
+    unsafe {
+        let v: u32 = A;
+        A = 0;
+        v
+    }
+}
```

Comment:
Is the logic in this PR sound in the presence of non-mut non-Freeze statics?  For example:

```Rust
use std::sync::atomic::{AtomicU32, Ordering};
static A: AtomicU32 = AtomicU32::new(42);

pub fn side_effect() -> u32 {
    let v: u32 = A.load(Ordering::Relaxed);
    A.store(0, Ordering::Relaxed);
    v
}
```

Reply:
No, it doesn't take `Freeze`-able things into account. I've filed an issue about this at #1957.